esphome/components/qmc5883l/sensor.py

```diff
@@ -52,6 +56,22 @@
 }
 
 
+def validate_config(config):
+    valErrors = []
+
+    # "never" is Translated to uint32_t MAX (4294967295)
+    if config[CONF_UPDATE_INTERVAL] in [4294967295, "never"]:
+        if CONF_DRDY_PIN not in config:
+            valErrors.append('"drdy_pin" REQUIRED for update_interval of "never"')
+        if config[CONF_DATA_RATE] is None:
+            valErrors.append('"data_rate" REQUIRED for update_interval of "never"')
+
+    if len(valErrors) > 0:
+        raise cv.Invalid("\n".join(valErrors))
+
```

Comment:
```suggestion
    if CONF_DATA_RATE not in config:
        config[CONF_DATA_RATE] = auto_data_rate(config)
```
Can just modify the config here instead of adding the if later.

Reply:
We want to modify something in a function called validate_config ???

Reply:
Sure

Reply:
You can rename it to validate_data_rate if you want.

Reply:
The other reason for doing this is for visibility. 

You are setting this config variable automatically. The user should be able to see this in the generated config when when they validate.

Reply:
It's only automatic if unspecified.

I apologize that these are difficult for me to follow.

Reply:
Updated the suggestion.

Reply:
It's only automatic if using update_interval with no drdy pin.